### PR TITLE
Migrations can be sent to all shards, without specifying them

### PR DESCRIPTION
**Pull request by adomokos**
We need this change for sharded migrations.

We need to add the macro ` using_all_shards` to follow Octopus' convention.
See the migration file in the file list.

### CODE REVIEW

File: lib/octopus.rb [hunk 1]
```diff
@@ -106,6 +106,12 @@ def self.rails5?
     ActiveRecord::VERSION::MAJOR == 5
   end
 
+  # Not all Octopus configuration might be in AR's Connection object
+  # This method will return what configured and visible to ActiveRecord
```
Comment (challeng) [inline lib/octopus.rb]: s/what/what is

File: lib/octopus/migration.rb [hunk 2]
```diff
@@ -24,6 +24,13 @@ def self.included(base)
     end
 
     module ClassMethods
+      def using_all_shards
+        return self unless connection.is_a?(Octopus::Proxy)
+
+        self.current_shard = Octopus.configured_shard_keys
```
Comment (challeng) [inline lib/octopus/migration.rb]: Could use `tap` and remove the `return self`

File: spec/migrations/16_create_user_on_all_shards.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateUserOnAllShards < ActiveRecord::Migration
+  using_all_shards
```
Comment (challeng) [inline spec/migrations/16_create_user_on_all_shards.rb]: So this is what we will be adding to all future Beekeeper migrations?

File: lib/octopus.rb [hunk 4]
```diff
@@ -106,6 +106,12 @@ def self.rails5?
...
+  def self.configured_shard_keys
+    ActiveRecord::Base.connection.instance_variable_get(:@shards).keys.map(&:to_sym)
```
Comment (challeng) [inline lib/octopus.rb]: Can we call this at all in Beehive? It seems like a nice helper method

File: lib/octopus/migration.rb (hunk 2 above)
Comment (adomokos) [reply lib/octopus/migration.rb]: That's what the other macros had. I tried to follow the convention believing that one day we might be able to submit this PR to Octopus.

**Pull request merged**